x-pack/plugins/security_solution/public/timelines/components/edit_data_provider/helpers.test.tsx

```diff
@@ -323,4 +384,37 @@ describe('helpers', () => {
       ).toBe(true);
     });
   });
+
+  describe('isValueFieldInvalid', () => {
+    it('returns false (not invalid) if DataProvierType type is template', () => {
+      expect(isValueFieldInvalid(DataProviderType.template, 'some string')).toBe(false);
+    });
+
+    it('returns false (not invalid) if DataProvierType type is default, and value is not template-like', () => {
+      expect(isValueFieldInvalid(DataProviderType.default, 'some string')).toBe(false);
+    });
+
+    it('returns true (invalid) if DataProvierType type is default, and value is template-like', () => {
+      expect(isValueFieldInvalid(DataProviderType.default, '{ oh nooo!')).toBe(true);
+      expect(isValueFieldInvalid(DataProviderType.default, 'oh nooo! }')).toBe(true);
+    });
+  });
+
+  describe('sanatizeValue', () => {
+    it("returns a provided value if it's a string or number as a string", () => {
+      expect(sanatizeValue('a string')).toBe('a string');
+      expect(sanatizeValue(1)).toBe('1');
+    });
+
+    it('returns the sting interpretation of the first value of an array', () => {
+      expect(sanatizeValue(['a string', 'another value'])).toBe('a string');
+      expect(sanatizeValue([1, 'another value'])).toBe('1');
+    });
+
+    it('handles odd array values', () => {
+      expect(sanatizeValue([])).toBe('undefined');
```

Comment:
Would an empty string be a valid sanitized value for an empty array (instead of `'undefined'`?)